### PR TITLE
TE-401 initial commit

### PR DESCRIPTION
**Pull request by gunther-bachmann**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+       '''
+    def lastCommit = readFile('last.commit.txt').trim()
+    return  lastCommit.startsWith('jenkins@ci.testeditor.org')
+}
+
+boolean isMaster() {
+    return env.BRANCH_NAME == 'master'
+}
+
+void withGradleEnv(List envVars = [], def body) {
```
Comment (franzbecker) [inline Jenkinsfile]: We should really extract this stuff and reuse it.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+language: java
```
Comment (franzbecker) [inline .travis.yml]: Not sure if we should use Travis here as we build it on the Jenkins anyway.

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    id 'net.researchgate.release' version '2.3.4'
+    id "com.jfrog.bintray" version "1.7"
+    id 'java-gradle-plugin'
+}
+
+repositories {
+    jcenter()
+}
+
+def actualVersion = "$version" // remapping version to this var, since version has namespace clashes in some cases (e.g. within bintray.pkg.version)
```
Comment (franzbecker) [inline build.gradle]: Use `project.version` instead - I think this should solve it without adding an additional variable.

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+def scmTag='v'+actualVersion
```
Comment (franzbecker) [inline build.gradle]: This is only used in the `bintray` section and should be declared there. Also `"v$project.version"` reads more nicely.

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+// for more configuration options see https://github.com/bintray/gradle-bintray-plugin
+bintray {
+    user = System.getenv()['JCENTER_USER'] 
+    key = System.getenv()['JCENTER_PASSWD']
+    publications = ['FixturePublication'] 
+    publish = true 
+    pkg {
+        repo = 'Fixtures'
```
Comment (franzbecker) [inline build.gradle]: Should we really maintain different Maven repositories? We could pack everything into `test-editor-maven`.
Or we could also create a new one called `maven` only.
Currently the url is `ttp://dl.bintray.com/test-editor/test-editor-maven` but could be `ttp://dl.bintray.com/test-editor/maven`

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        name = 'core-fixture'
```
Comment (franzbecker) [inline build.gradle]: This should be the project's name

File: build.gradle [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+release {
+    tagTemplate = 'v${version}'
+    preTagCommitMessage = '[release] '
+    tagCommitMessage = '[release] '
+    newVersionCommitMessage = '[new version]'
+
+    scmAdapters = [
+        net.researchgate.release.GitAdapter
```
Comment (franzbecker) [inline build.gradle]: This should not be necessary.

File: build.gradle [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    ]  
+    git {
```
Comment (franzbecker) [inline build.gradle]: This should not be necessary.

File: gradle/wrapper/gradle-wrapper.properties [hunk 9]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://downloads.gradle.org/distributions/gradle-2.14-bin.zip
```
Comment (franzbecker) [inline gradle/wrapper/gradle-wrapper.properties]: We should use Gradle 2.14.1

File: build.gradle [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+        requireBranch = 'master'
+        pushToRemote = 'origin'
+    }
+}
+
+dependencies {
+    afterReleaseBuild.dependsOn bintrayUpload
```
Comment (franzbecker) [inline build.gradle]: This should be removed. And why is it in dependencies?

File: build.gradle (hunk 8 above)
Comment (franzbecker) [inline build.gradle]: It clutters up the build and someone familiar with the plugin wonders if the values are different from the default values. So one needs to read those as well.

File: build.gradle (hunk 10 above)
Comment (gunther-bachmann) [inline build.gradle]: got that notion from https://github.com/researchgate/gradle-release#custom-release-steps

**Pull request merged**